website/docs/proposals/20250630-kaito-cli.md

```diff
@@ -0,0 +1,526 @@
+
+# Deploy with specific runtime
+kubectl kaito deploy --name my-workspace --model phi-3-mini --runtime vllm
+kubectl kaito deploy --name my-workspace --model falcon-7b --runtime transformers
+
+# Deploy with adapters
+kubectl kaito deploy --name my-workspace --model phi-3-mini \
+  --adapter my-adapter:v1 --adapter-strength 0.8
+
+# Deploy distributed inference
+kubectl kaito deploy --name my-distributed --model llama-3.3-70b \
+  --distributed --nodes 4 --runtime vllm
+
+# Deploy with feature gates
+kubectl kaito deploy --name my-workspace --model phi-3-mini --feature-gates vLLM=true,ensureNodeClass=false
+```
+
+#### Management Commands
+
+```bash
+# Scale inference replicas
+kubectl kaito scale workspace/my-workspace --replicas 3
+
+# Status and monitoring
+kubectl kaito status workspace/my-workspace
+kubectl kaito logs workspace/my-workspace --follow
+
+# Health checks
+kubectl kaito health check --workspace my-workspace --detailed
+```
+
+### Tuning Workflows
+
+#### Basic Tuning
+
+```bash
+# Deploy tuning workload
+kubectl kaito tune --name my-tuning --model phi-3-mini --method qlora --input gs://my-dataset --output my-registry/adapter:v1
+
+# Advanced tuning with custom configuration
+kubectl kaito tune --name my-tuning --model falcon-7b --method lora \
+  --config my-lora-config \
+  --input pvc://training-data \
+  --output pvc://output-adapters \
+  --epochs 5 \
+  --batch-size 4
+```
+
+#### Tuning Management
+
+```bash
+# Tuning status and monitoring
+kubectl kaito tune status my-tuning
+kubectl kaito tune logs my-tuning --follow
+kubectl kaito tune metrics my-tuning --show-gpu-utilization
+
+# Tuning with resource optimization
+kubectl kaito tune --name my-tuning --model llama-3.3-70b --method qlora \
+  --auto-optimize-resources \
+  --instance-type Standard_NC24ads_A100_v4 \
+  --show-cost-estimate
+```
+
+### RAG Operations
+
+#### Deployment
+
+```bash
+# Deploy RAG engine with local embedding
+kubectl kaito rag deploy --name my-rag --embedding-model BAAI/bge-small-en-v1.5 --inference-url http://my-service/v1/completions
+
+# Deploy complete RAG stack with workspace integration
+kubectl kaito rag deploy --name my-rag-app \
+  --embedding-model BAAI/bge-small-en-v1.5 \
+  --llm-workspace my-inference-workspace \
+  --vector-store faiss \
+  --instance-type Standard_NC6s_v3
+
+# Deploy RAG with external inference service
+kubectl kaito rag deploy --name my-rag-external \
+  --embedding-model BAAI/bge-small-en-v1.5 \
+  --inference-url https://api-inference.huggingface.co/models/HuggingFaceH4/zephyr-7b-beta/v1/completions \
+  --inference-secret huggingface-token
+```
+
+#### Document and Query Management
+
+```bash
+# Manage RAG indices and documents
+kubectl kaito rag index --name my-rag --add-document doc.pdf
+kubectl kaito rag index --name my-rag --add-documents-dir ./docs --batch-size 10
+kubectl kaito rag index --name my-rag --list-documents
+kubectl kaito rag index --name my-rag --delete-document doc-id-123
+
```

Comment:
For this use case, we can consider generating a cli based on openapi spec.
It would be easier to maintain.